R/build_factory.R

```diff
@@ -75,19 +110,31 @@ build_factory <- function(fun,
     )
   })
 
-  if (pass_dots) {
+  if (.pass_dots) {
     args <- rlang::pairlist2(
       !!!args,
       "..." =
       )
     old_fun <- fun
-    body(fun) <- body_insert(
-      fn_body = body(fun),
-      insertion = quote(!!!dots),
-      before = quote(...)
-    )
+
+    # If they want to pass ... to the child, we *add* !!!dots. Else we replace
+    # ... with !!!dots.
+    if ("..." %in% names(formals(fun))) {
```

Comment:
personal preference, but you could also use `formalArgs(fun)` in place of `names(formals(fun))`

Reply:
Huh, I didn't know about that one, thanks! That said... it doesn't really save many characters, and does the same thing, so I think I'll just leave it as-is.